books/book.js

```diff
@@ -0,0 +1,40 @@
+let currentUrl = new URL(location.href); // creating new url that contains current url
+let searchParams = new URLSearchParams(currentUrl.search); //  searching in current  url params
+let currentUrlId = searchParams.get("id"); //  getting current url id// Fetching book when page is loaded
+window.addEventListener("DomContentLoaded", getBook());
+
+function getBook() {
+  fetch(`https://bookshop-api.mirkwood.dev/books/${currentUrlId}`)
+    .then((res) => {
+      return res.json();
+    })
+
+    .then((book) => {
+      displayBook(book);
+    });
+}
+
+function displayBook(book) {
+  let main = document.querySelector("main"); // html main element
+  let currentBook = "";
+  let bookID = book.id; // getting id of current book
+  // comparing current book's id with current  url id
+
+  // Getting values of online API and adding them to the UI
+  let title = ` <h2 class = 'title' > ${book.title} </h2>`;
+  let author = `<h5 class = 'author'> By (author)  ${book.author.first_name} ${book.author.last_name}</h5> `;
+  let bookYear = `<h5 class = 'book-year'> Originally published: ${book.year} </h5>`;
+  let topic = `<h5 class = 'topic'> Topic: ${book.topics[0].name} </h5>`;
+  let synopsis = `<p class = 'synopsis'> ${book.synopsis} </p> `;
+  let bookPrice = `<h2> ${book.price} € </h2>`;
+  let bookPurchaseH3 = `<h3> Free delivery worldwide</h3>`;
+  let bookPurchaseP = `  <p> <i class="far fa-check-circle"></i> Available. Dispatched from Australia in 3  business  days. </p>`;
+  let buttonBasket = `<button class = 'basket' > Add to basket </button>`;
+  let buttonWish = `<button class = 'wishlist' > Add to wishlist </button>`;
```

Comment:
Since we will buy a single book from the details page, you could add another "Buy now" button. When this button is clicked, a call to the REST API purchase endpoint should be made, but that's another chunk of work.

For now just add the button :)

Reply:
Done this